transitclock/src/main/java/org/transitclock/core/AvlProcessor.java

```diff
@@ -444,9 +444,13 @@ public void matchNewFixForPredictableVehicle(VehicleState vehicleState) {
 		// Find possible spatial matches
 		List<SpatialMatch> spatialMatches = SpatialMatcher
 				.getSpatialMatches(vehicleState);
+		
+		// This will augment the results of the original spatial matcher with the match from barefoot
+		spatialMatches.add(vehicleState.getMapMatchedSpatialMatch());

```

Comment:
It may be good to record the provenance of the match in spatialMatches that ultimately gets selected in getBestTemporalMatch